dashboard/src/components/ErrorBoundary/ErrorBoundary.test.tsx

```diff
@@ -0,0 +1,62 @@
+import { mount } from "enzyme";
+import * as React from "react";
+import ErrorBoundary from ".";
+import { UnexpectedErrorAlert } from "../ErrorAlert";
+
+describe("ErrorBoundary around a component", () => {
+  const exampleError = new Error("Bang!");
+  const BadRenderor = (props: { throwError?: boolean }) => {
+    if (props.throwError) {
+      throw exampleError;
+    }
+    return <div className="no-error" />;
+  };
+
+  // tslint:disable:no-console
+  beforeEach(() => {
+    // To avoid polluting the logs
+    console.error = jest.fn();
```

Comment:
### More Granular mocking

That's a good point. But I am not sure that change is worth doing here because:

 In the first test we need to mock console.log before we mount the wrapper (since the wrapper calls console.log if there is an error and we do not want to pollute the console).

In the second test I can do what you said.

So the code would look like something:

```diff
git diff
diff --git a/dashboard/src/components/ErrorBoundary/ErrorBoundary.test.tsx b/dashboard/src/components/ErrorBoundary/ErrorBoundary.test.tsx
index 7c46cb1f..76bd8788 100644
--- a/dashboard/src/components/ErrorBoundary/ErrorBoundary.test.tsx
+++ b/dashboard/src/components/ErrorBoundary/ErrorBoundary.test.tsx
@@ -13,12 +13,11 @@ describe("ErrorBoundary around a component", () => {
   };
 
   // tslint:disable:no-console
-  beforeEach(() => {
-    // To avoid polluting the logs
-    console.error = jest.fn();
-  });
 
   it("captures any synchronous error thrown during a descendant render", () => {
+    // To avoid polluting the logs and check that it was called
+    console.error = jest.fn();
+
     const wrapper = mount(
       <ErrorBoundary>
         <BadRenderor throwError={true} />
@@ -55,7 +54,8 @@ describe("ErrorBoundary around a component", () => {
 
     // the state is null
     expect(wrapper.state()).toEqual({ error: null, errorInfo: null });
-
+    // To avoid polluting the logs and check that it was called
+    console.error = jest.fn();
     // console.error is not called
     expect(console.error).not.toHaveBeenCalled();
   });

```

Which still means that in theory we have pretty much the bulk of a test case ran with a mocked console.log.

Also, note that even having console.log mocked, errors are still shown in the console since it uses the jest logger. 

```jest
 FAIL  src/components/ErrorBoundary/ErrorBoundary.test.tsx
  ErrorBoundary around a component
    ✕ captures any synchronous error thrown during a descendant render (55ms)
    ✓ renders only the wrapped components if no error (5ms)

  ● ErrorBoundary around a component › captures any synchronous error thrown during a descendant render

    ReferenceError: thisDoesNotExist is not defined

      35 |     const errorInfo: React.ErrorInfo = wrapper.state("errorInfo");
      36 |     expect(errorInfo.componentStack.length).not.toEqual(0);
    > 37 |     thisDoesNotExist
```

### Restoring mock

In theory, jest should not propagate mocks outside of the files, and I actually thought that it should not do it outside of the `describe` but it does. So I'll add a restore. Thanks!